File: 26069/26069.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    26069.py                                           :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2024/03/17 00:29:12 by jihoolee          #+#    #+#              #
#    Updated: 2024/03/17 00:32:00 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

def main() -> None:
    rainbow_deacers: set = set(['ChongChong'])

    for _ in range(int(input())):
        name1, name2 = input().split()
        if name1 in rainbow_deacers:
            rainbow_deacers.add(name2)
        elif name2 in rainbow_deacers:
            rainbow_deacers.add(name1)
    print(len(rainbow_deacers))


if __name__ == '__main__':
    main()

File: 26069/26069.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   26069.cpp                                          :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2024/03/17 00:20:52 by jihoolee          #+#    #+#             */
/*   Updated: 2024/03/17 00:27:18 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <string>
#include <set>

int main(void) {
  int                   input_cnt;
  std::set<std::string> rainbow_dancers;
  std::string           name1, name2;

  rainbow_dancers.insert("ChongChong");
  std::cin >> input_cnt;
  for (int i = 0; i < input_cnt; ++i) {
    std::cin >> name1 >> name2;
    if (rainbow_dancers.find(name1) != rainbow_dancers.end())
      rainbow_dancers.insert(name2);
    else if (rainbow_dancers.find(name2) != rainbow_dancers.end())
      rainbow_dancers.insert(name1);
  }
  std::cout << rainbow_dancers.size();
  return 0;
}
